platforms/documentation/docs/src/docs/userguide/dep-man/advanced/artifact_resolution.adoc

```diff
@@ -0,0 +1,356 @@
+The `runtimeElements` *variant* of `org.jetbrains.kotlin:kotlin-stdlib:1.8.10` provides three *artifact sets*—`jar`, `classes`, and `resources`—each representing the same distributable in a different form.
+
+Gradle must now follow a specific process to determine the most appropriate *artifact set* for the build.
+
+[[sec:artifact-selection]]
+== Artifact Resolution Flow
+
+*Artifact selection* operates on the dependency graph on a __node-by-node__ basis.
+
+Each _node_ (i.e., *variant*) in the graph may expose multiple sets of *artifacts*, but only one of those sets may be selected.
+
+image::dep-man-adv-4.png[]
+
+For each _node_, or *variant*, in a graph, Gradle performs *attribute* matching over each set of *artifacts* exposed by that _node_ to determine the best *artifact set*.
+
+If no *artifact sets* match the requested *attributes*, Gradle will attempt to construct an <<artifact_transforms.adoc#artifact-transforms,artifact transform>> chain to satisfy the request.
+
+For more details on the *attribute* matching process, see the previous <<variant_aware_resolution.adoc#sec:understanding-variant-selection,section>>.
+
+== Available APIs
+
+Gradle APIs can be used to influence the process of *artifact selection*.
+
+Gradle can then expose the results of artifact selection as an `ArtifactCollection`.
+More commonly, the results are exposed as a `FileCollection`, which is a flat list of files.
+
+[[sec:implicit-artifact-selection]]
+=== Implicit Artifact Selection
+
+By default, the *attributes* used for *artifact* selection are the same as those used for *variant* selection during graph resolution.
+These *attributes* are specified by the link:{javadocPath}/org/gradle/api/attributes/HasAttributes.html#getAttributes()[`Configuration.getAttributes()`] property.
+
+To perform *artifact* selection (and implicitly, graph resolution) using these default *attributes*, use the `FileCollection` and `ArtifactCollection` APIs.
+
+NOTE: Files can also be accessed from the configuration's `ResolvedConfiguration`, `LenientConfiguration`, `ResolvedArtifact` and `ResolvedDependency` APIs.
+However, these APIs are in maintenance mode and are discouraged for use in new development.
+These APIs perform artifact selection using the default attributes.
+
+==== Resolving files
+
+To resolve files, we first define a task that accepts a `ConfigurableFileCollection` as input:
+
+====
+include::sample[dir="snippets/dependencyManagement/performingResolution-artifactResolution/kotlin",files="build.gradle.kts[tags=file-resolution-task]"]
+include::sample[dir="snippets/dependencyManagement/performingResolution-artifactResolution/groovy",files="build.gradle[tags=file-resolution-task]"]
+====
+
+Then, we can wire up a resolvable configuration's files to the task's input.
+The `Configuration` directly implements `FileCollection` and can be wired directly.
+Alternatively, wiring through `Configuration.getIncoming()` is a more explicit approach:
```

Comment:
💭 I think we should only provide one example here.  Multiple ways to do things is such a large complaint about Gradle.

Reply:
Fixed.